### DISCUSSION TITLE
How to send templated Whatsapp messages via API?

### DISCUSSION THREAD

**Discussion by rolymaz** [chatwoot/chatwoot#7891] [Chatwoot APIs]
Can someone confirm the basic steps to send a templated Whatsapp messages via API.

With reference to these guides, 

https://github.com/chatwoot/chatwoot/issues/5958
https://www.chatwoot.com/docs/product/channels/api/send-messages
https://github.com/chatwoot/chatwoot/issues/6704 (Link is broken) 

As I understand it: 

Step 1. Create the contact
https://www.chatwoot.com/developers/api/#tag/Contacts/operation/contactCreate

Step 2. Create the conversation (using contact_id from Step 1)
https://www.chatwoot.com/developers/api/#tag/Conversations/operation/newConversation

Step 3. Create the message
https://www.chatwoot.com/developers/api/#tag/Messages/operation/create-a-new-message-in-a-conversation
With message payload:
"content": "WRITE YOUR CONTENT HERE", "template_params": { "name": "template name", "category": "TRANSACTIONAL", "language": "en_US", "processed_params": { "1": "write your field value here", "2": "Write your field value here" } }

**Comment by rolymaz** [comment]
Has anyone been successful in using the APIs to sent out templated WA messages to new customers?

**Comment by ghost** [comment]
Also, has anyone been able to send message templates with buttons, header and buttons, all these components with parameters to WhatsApp through Chatwoot API?

**Reply by glauberssoutodossantos** [reply]
Hello,
Using the chatwooot APIs, it is not possible to use templates with headers. Buttons have been available for some time, using interactive messages.

**Comment by ghost** [comment]
> Hello, Using the chatwooot APIs, it is not possible to use templates with headers. Buttons have been available for some time, using interactive messages.

Any docs or examples?

**Reply by Shiba1805** [reply]
Any docs or exemples?

**Comment by rolymaz** [comment]
Here are some example API calls and payloads, to send a whatsapp message via API,  courtesy discord: 

https://discord.com/channels/647412545203994635/1148618372146274415/1148619067289260134

**Comment by michaelthedev** [comment]
Hi, for anybody that finds this. I just finished setting this up myself

Creating the contact is pretty straightforward.
Step 3 from above is not needed because you can send an initial message while creating the conversation.

```json
{
  "source_id": "**source id from create contact**",
  "inbox_id": "**inbox_id**",
  "contact_id": "**id of the contact you just created**",
  "status": "open",
  "message": {
    "content": "**Message content here and edit params if needed**",
    "template_params": {
      "name": "template_name",
      "category": "MARKETING",
      "language": "nl",
      "processed_params": {
        "name": "Michael"
      }
    }
  }
}
```
